README.md

```diff
@@ -1,72 +1,93 @@
-9. in the `testenv2/conf/sp_metadata.xml` file:
-  - remove the `<EncryptionMethod>` elements
-  - between the elements `<KeyDescriptor>` and `<NameIDFormat>` add the following element:
-     ```
-     <SingleLogoutService 
-         Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST"
-         Location="http://io-onboarding-backend:3000/spid/logout" /> 
-    ``` 
-10. run `docker-compose restart` to restart the containers
-11. point your browser to [http://io-onboarding-backend:3000](http://io-onboarding-backend:3000)
-
-### SPID test login
-All the available SPID test users with their login credentials and attributes are listed in the page [http://spid-testenv2:8088/users](http://spid-testenv2:8088/users), you can add more if needed. 
-There are two default available users whose login credentials are respectively the following:
-  - username `pippo` and password `test`
-  - username `pinco` and password `test`
-
-In order to perform a SPID test login, point your browser to [http://io-onboarding-backend:3000/login?entityID=xx_testenv2&authLevel=SpidL2](http://io-onboarding-backend:3000/login?entityID=xx_testenv2&authLevel=SpidL2) and use the login credentials of one of the SPID test users. 
- 
-
-### Environment variables
-
-The table below describes all the Environment variables needed by the application.
+A local SPID test environment is needed to fully test the application features. [spid-testenv2](https://github.com/italia/spid-testenv2) has been selected to achieve this goal.
+
+### Build steps
+
+The *Dockerfile* and the `docker-compose.yaml` files used to build the application, are in the root of this repository.
+
+To build the application, run from the project root:
+
+```shell
+docker-compose up -d
+```
+
+>NOTE: the *docker-compose.yaml* file sets some environment variables that could be used to adapt the application features to specific needs. Variable values can be modified editing the *.env.example* file in this repository. More info about variables can be found in the dedicated paragraph, below in the readme.
+
+Backend REST APIs can be now accessed at [http://localhost:3000](http://localhost:3000).
+
+To bring down the test environment and remove the containers use
+
+```shell
+docker-compose down
+```
+
+Sometimes, you may need to rebuild the software. To do so, make sure you re-build the backend container, typing:
+
+```shell
+docker-compose up --build
+```
+
+### Modify SPID Service Provider certificates
+
+It may be sometimes needed to modify the default SPID Service Provider certificates that come with the application.
+
+To create new certificates, run the command 
+
+```shell
+yarn generate-test-certs
+```
+
+Then, start the containers using *docker-compose* and point your browser to [http://localhost:3000/metadata](http://localhost:3000/metadata).
+You should now manually replace the content of the *testenv2/conf/sp_metadata.xml* file with the output of this page.
```

Comment:
I have a doubt: could `the output` be ambiguous? Is there a chance that the user could copy the rendered text from the browser instead of the source of the page from the body response?

Reply:
I think it's clear enough..